### DISCUSSION TITLE
package "panelvar" in R _ error

### DISCUSSION THREAD

**Discussion by la-guardia-zorzin** [community/community#110983] [Packages]
Please, I'm facing an error when running a PVAR model in R, using the "panelvar" package.

The error is:  "Errore cholmod 'problem too large' nel file ../Core/cholmod_dense.c, linea 102"

Below, my code for the model... My base is made up of 2,618 observations, so I don't believe it's a memory problem.

varone <-pvargmm(
  dependent_vars = c("Img","d_Y_Pop_c","d_Y_Pop_I", "GDP_c", "GDP_I", "d_Unemploy_c", "d_Unemploy_I"),
  lags = 1,
  exog_vars = c("Policy_c", "Policy_I", "W_disaster_c", "W_disaster_I"),
  transformation = "fd",
  data = df2,
  panel_identifier = c("Time", "Country"),
  steps = c("twostep"),
  system_instruments = TRUE,
  max_instr_dependent_vars = 99,
  min_instr_dependent_vars = 2L,
  collapse = FALSE
)

Thank you very much in advance for any help (or any clue, any guidance)!!! 

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ardagurkan9** [comment]
The issue is likely caused by the order of your panel identifiers.

In `panelvar`, the first variable should be the panel unit and the second should be the time variable. So try changing:

```r
panel_identifier = c("Time", "Country")
```

to:

```r
panel_identifier = c("Country", "Time")
```

Also, your instrument setup is quite large. Using `max_instr_dependent_vars = 99`, `collapse = FALSE`, and system GMM can create a very large instrument matrix even with only 2,618 observations.

I would also try:

```r
max_instr_dependent_vars = 4L,
collapse = TRUE
```

If it still fails, test the model with `steps = "onestep"` and `system_instruments = FALSE` first, then add them back one at a time.
